app/scripts/controllers/statefulSet.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+// TODO: statefulSetController rename
+// Kube docs: http://kubernetes.io/docs/user-guide/statefulSet/
+angular
+  .module('openshiftConsole')
+  .controller('StatefulSetController', function(
+    $filter,
+    $scope,
+    $routeParams,
+    AlertMessageService,
+    BreadcrumbsService,
+    DataService,
+    ProjectsService) {
+
+    $scope.projectName = $routeParams.project;
+    $scope.statefulSetName = $routeParams.statefulset || $routeParams.petset; // TODO: eliminate petset fallback
+    $scope.forms = {};
+    $scope.alerts = {};
+    // TODO: update getBreadcrumbs, currently returns:
+    //  Pet Sets > hello-statefulSet
+    // But the link for `Pet Sets` isn't valid. Where do we want these to live?
+    $scope.breadcrumbs = BreadcrumbsService.getBreadcrumbs({
+      name: $scope.statefulSetName,
+      kind: 'statefulSet', // TODO: update to statefulSet
+      namespace: $routeParams.project
+    });
+    $scope.emptyMessage = "Loading...";
+
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+
+    var watches = [];
+    var projectContext;
+
+    var updatePods = function(pods, selector) {
+      if (!pods || !selector) {
+        return;
+      }
+      return _.filter(pods, function(pod) {
+        return selector.matches(pod);
+      });
+    };
+
+    var resourceGroupVersion = {
+      resource: 'petsets',// TODO: statefulSets
+      group: 'apps',
+      version: 'v1alpha1'
+    };
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        projectContext = context;
+
+        watches.push(DataService.watchObject(resourceGroupVersion, $scope.statefulSetName, context, function(statefulSet) {
+          console.log('statefulSet', statefulSet);
+          angular.extend($scope, {
+            resourceGroupVersion: resourceGroupVersion,
+            statefulSet: statefulSet,
+            loaded: true,
+            projectContext: context
+            // TODO: support scaling
+            // scale: function() {},
+            // isScalable: function() { return true; }
+          });
+          var pods;
+          var selector;
+          $scope.$watch('statefulSet.spec.selector', function() {
+            selector = new LabelSelector($scope.statefulSet.spec.selector);
+            $scope.podsForStatefulSet = updatePods(pods, selector);
+          }, true);
+          // TODO: this is leaky. every time the outer watch updates,
+          // we are creating a new watch on pods:
+          // - console.log('# of watches?', watches.length); // steadily increments
```

Comment:
Oh right, and I did all those PRs to share the callback fn a while back.  Updated.